block_my_feedback.php

```diff
@@ -168,6 +185,53 @@ public function fetch_feedback() : array {
         return  $template->feedback;
     }
 
+    /**
+     * Checking Review options for showing quiz submissions.
+     *
+     * @param $submission
+     * @return bool
+     * @throws dml_exception
+     */
+    public function show_quiz_submission($submission): bool {
+        global $DB;
+        $settings = $DB->get_record('quiz', ['id' => $submission->instance]);
+
+        // The Quiz Review options are represented each by a combined hexadecimal value stored as a decimal integer.
```

Comment:
This smells bad :(
If there is a way of getting this from the quiz mode that would feel less fragile than redefine it all again locally - in case it changes.

Reply:
@opitz May be you could have a look at https://github.com/moodle/moodle/blob/main/mod/quiz/classes/question/display_options.php#L70
I think we can get the quiz's options from there and then do the checking.

For example:
`
$openreviewoptions = display_options::make_from_quiz($quizobj, display_options::LATER_WHILE_OPEN);
$closedreviewoptions = display_options::make_from_quiz($quizobj, display_options::AFTER_CLOSE);
`

Reply:
@stuartlamour @aydevworks Thanks for the feedback.
I have now implemented a more elegant way of dealing with review options - thanks Alex!